File: Java/1751-1800/1758. Minimum Changes To Make Alternating Binary String.java
// https://leetcode.com/problems/minimum-changes-to-make-alternating-binary-string/

class Solution {
    public int minOperations(String s) {
        boolean binary = true;
        int zero = 0, one = 0;

        for(char letter : s.toCharArray()){
            if(binary){
                if(letter == '0') one++;
                else zero++;
            }else{
                if(letter == '0') zero++;
                else one++;
            }
            binary = !binary;
        }

        return Math.min(one, zero);
    }
}

File: Python/1751-1800/1758. Minimum Changes To Make Alternating Binary String.py
# https://leetcode.com/problems/minimum-changes-to-make-alternating-binary-string/

class Solution:
    def minOperations(self, s: str) -> int:
        binary = True
        zero, one = 0, 0

        for letter in s:
            if binary :
                if letter == '0':
                    one += 1
                else:
                    zero += 1
            else:
                if letter == '0':
                    zero += 1
                else:
                    one += 1
            binary = not binary

        return min(one, zero)
